### PR TITLE
Lineage UI: Add a graph depth config option

### PR DESCRIPTION
**Pull request by jlukenoff**
### Problem
Currently for very large lineage graphs, it can be hard to navigate the graph by dragging and zooming.

Closes: #2378 

### Solution

This PR adds an option to the lineage graph UI where the user can select the depth of the lineage graph.

### Checklist

- [x] You've [signed-off](https://github.com/MarquezProject/marquez/blob/main/CONTRIBUTING.md#sign-your-work) your work
- [ ] Your changes are accompanied by tests (_if relevant_)
- [x] Your change contains a [small diff](https://kurtisnusbaum.medium.com/stacked-diffs-keeping-phabricator-diffs-small-d9964f4dcfa6) and is self-contained
- [ ] You've updated any relevant documentation (_if relevant_)
- [x] You've included a one-line summary of your change for the [`CHANGELOG.md`](https://github.com/MarquezProject/marquez/blob/main/CHANGELOG.md#unreleased) (_Depending on the change, this may not be necessary_).
- [ ] You've versioned your `.sql` database schema migration according to [Flyway's naming convention](https://flywaydb.org/documentation/concepts/migrations#naming) (_if relevant_)
- [x] You've included a [header](https://github.com/MarquezProject/marquez/blob/main/CONTRIBUTING.md#copyright--license) in any source code files (_if relevant_)

### CODE REVIEW

**Comment by jlukenoff** [comment]
I just added this as floating in the upper right-hand corner for now. Happy to move it wherever it makes the most sense, although this feels pretty usable to me

output

File: web/src/store/reducers/lineage.ts [hunk 1]
```diff
@@ -5,26 +5,31 @@ import {
...
   bottomBarHeight: number
+  depth: number
 }
 
 const initialState: ILineageState = {
   lineage: { graph: [] },
   selectedNode: null,
-  bottomBarHeight: (window.innerHeight - HEADER_HEIGHT) / 3
+  bottomBarHeight: (window.innerHeight - HEADER_HEIGHT) / 3,
+  depth: 1
```
Comment (phixMe) [inline web/src/store/reducers/lineage.ts:27]: Can we make this default to a larger number that would render the whole demo graph at least?

Comment (jlukenoff) [reply web/src/store/reducers/lineage.ts]: Good point, updated it to `5`. Trying to bias towards a lower value as, at least in my experience playing around with the demo, it feels less clunky to slowly increase from a lower number to "zoom out" vs. walking down from a higher number to "zoom in".

**Review by phixMe — APPROVED** [review]
Thanks @jlukenoff 🐑 it!

**Comment by jlukenoff** [comment]
Thanks @phixMe ! Just added an entry to the changelog. Would you mind merging? I don't have write access yet

**Review by wslulciuc — APPROVED** [review]
Love it! Let's favor usability for now and 🚢 it as is!

**Pull request merged**